concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -709,14 +714,38 @@ impl ConsensusContainer {
         ))
     }
 
-    pub fn in_baking_committee(&self) -> ConsensusIsInBakingCommitteeResponse {
-        wrap_c_committee_call!(self, |consensus| bakerIdBestBlock(consensus))
+    /// Gets baker status of the node along with the baker ID registered in the
+    /// baker credentials used, if available.
+    pub fn in_baking_committee(&self) -> (ConsensusIsInBakingCommitteeResponse, Option<u64>) {
+        let consensus = self.consensus.load(Ordering::SeqCst);
+        let mut baker_id: u64 = 0;
+        let mut has_baker_id: u8 = 0;
+
+        let result = unsafe { bakerStatusBestBlock(consensus, &mut baker_id, &mut has_baker_id) };
+
+        let status = ConsensusIsInBakingCommitteeResponse::try_from(result).unwrap_or_else(|err| {
+            unreachable!("An error occured when trying to convert FFI return code: {}", err)
+        });
+
+        let baker_id_option = match has_baker_id {
+            0 => None,
+            1 => Some(baker_id),
+            code => unreachable!("Expected value from FFI to be either 0 or 1, received {}", code),
+        };
```

Comment:
Why not just treat 0 as false and anything else as true?

Reply:
Well, I took the logic from the macro that wraps ffi bool return calls, where this was the way it was done. And I also think it's nice to throw an error, since clearly someone has misunderstood the purpose of the value, if anything other than 0/1 is used.

Reply:
In that case, can you update the documentation on `bakerStatusBestBlock` to make the values explicit.